karapace/schema_models.py

```diff
@@ -180,6 +183,50 @@ def schema(self) -> Draft7Validator | AvroSchema | ProtobufSchema:
         return parsed_typed_schema.schema
 
 
+class AvroMerge:
+    def __init__(self, schema_str: str, dependencies: Mapping[str, Dependency] | None = None):
+        self.schema_str = json_encode(json_decode(schema_str), compact=True, sort_keys=True)
+        self.dependencies = dependencies
+        self.unique_id = 0
+
+    def union_safe_schema_str(self, schema_str: str) -> str:
+        # in case we meet union - we use it as is
+        regex = re.compile(r"^\s*\[")
+        if regex.match(schema_str):
+            return (
+                "{"
+                + f' "name": "___un__ique_x_n_q_karapace____{self.unique_id}",'
+                + '"type":"record","fields":[{"name":"name", "type":'
+                + schema_str
+                + "}]}"
+            )
+
+        return (
```

Comment:
Minor, but personally I think these Strings might be more readable like this (or something similar):

```python
        if regex.match(schema_str):
            return (
                f"{{ \"name\": \"___un__ique_x_n_q_karapace____{self.unique_id}\","
                f"\"type\":\"record\",\"fields\":[{{\"name\":\"name\", \"type\":{schema_str}}}]}}"
            )

        return (
            f"{{ \"name\": \"___un__ique_x_n_q_karapace____{self.unique_id}\",\"type\":\"record\","
            f"\"fields\":[{{\"name\":\"name\", \"type\": [{schema_str}]}}]}}"
        )
```

Reply:
Do You mean to make the field `name` more readable?

Reply:
No, i meant to make the returned strings a bit more readable, rather than being split with multiple `+`